integration-libs/opf/base/core/facade/opf-payment.service.ts

```diff
@@ -24,18 +53,130 @@ export class OpfPaymentService implements OpfPaymentFacade {
+        screenWidth: this.winRef.nativeWindow?.screen?.width,
+        userAgent: this.winRef.nativeWindow?.navigator?.userAgent,
+        originUrl: this.winRef.nativeWindow?.location?.origin,
+        timeZoneOffset,
+      },
+    };
+    if (paymentMethod !== PaymentMethod.CREDIT_CARD) {
+      submitRequest.encryptedToken = '';
+    }
+
+    return combineLatest([
+      this.userIdService.getUserId(),
+      this.activeCartFacade.getActiveCartId(),
+    ]).pipe(
+      switchMap(([userId, activeCartId]: [string, string]) => {
+        submitRequest.cartId = activeCartId;
+        return this.opfOtpFacade.generateOtpKey(userId, activeCartId);
+      }),
+      filter((response) => Boolean(response?.value)),
+      take(1),
+      concatMap(({ value: otpKey }) =>
+        this.opfPaymentConnector.submitPayment(
+          submitRequest,
+          otpKey,
+          paymentSessionId
+        )
+      ),
+      concatMap((response: SubmitResponse) => {
+        if (
+          response.status === SubmitStatus.ACCEPTED ||
+          response.status === SubmitStatus.DELAYED
+        ) {
+          return from(Promise.resolve(submitSuccess(response))).pipe(
+            concatMap(() => this.opfOrderFacade.placeOpfOrder(true))
+          );
+        } else if (response.status === SubmitStatus.PENDING) {
+          return from(Promise.resolve(submitPending(response))).pipe(
+            concatMap(() => EMPTY)
+          );
+        } else if (response.status === SubmitStatus.REJECTED) {
+          return from(Promise.resolve(submitFailure(response))).pipe(
+            concatMap(() =>
+              throwError({ ...this.defaultError, type: 'PAYMENT_REJECTED' })
+            )
+          );
+        } else {
+          return from(Promise.resolve(submitFailure(response))).pipe(
+            concatMap(() =>
+              throwError({
+                ...this.defaultError,
+                type: 'STATUS_NOT_RECOGNIZED',
+              })
+            )
+          );
+        }
+      }),
+      tap((order: Order) => {
+        if (order) {
+          this.routingService.go({ cxRoute: 'orderConfirmation' });
+        }
+      }),
+      map((order: Order) => (order ? true : false)),
+      catchError((error: PaymentError | undefined) => {
+        this.handlePaymentError(error, returnPath);
+        return throwError(error);
+      })
+    );
+  });
+
   constructor(
     protected queryService: QueryService,
     protected commandService: CommandService,
-    protected opfCheckoutConnector: OpfPaymentConnector
+    protected opfPaymentConnector: OpfPaymentConnector,
+    protected winRef: WindowRef,
+    protected opfOtpFacade: OpfOtpFacade,
+    protected activeCartFacade: ActiveCartFacade,
+    protected userIdService: UserIdService,
+    protected routingService: RoutingService,
+    protected opfOrderFacade: OpfOrderFacade,
+    protected globalMessageService: GlobalMessageService
   ) {}
 
+  defaultError: PaymentError = {
```

Comment:
I'm thinking about separate file for this declaration, we can call it `payment-error.model.ts` and put this as an export constant. Then here what we need to do is just import such variable if needed.

Reply:
yes,I though about it as it is stuffing the file. It will in root folder so VerificationPage can use it too.